atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/CellsSweeper.java

```diff
@@ -51,11 +91,49 @@ public void sweepCells(
         for (Follower follower : followers) {
             follower.run(txManager, tableRef, cellTsPairsToSweep.keySet(), Transaction.TransactionType.HARD_DELETE);
         }
+
         if (!sentinelsToAdd.isEmpty()) {
             keyValueService.addGarbageCollectionSentinelValues(
```

Comment:
Yep. The sentinels will only be written below some other value (the most recent value for the cell). After the restore, future reads will read this value and not the sentinel.